File: Solutions/Python/remove_duplicates_from_sorted_array_ii.py
"""
    [MEDIUM]
    80. Remove Duplicates from Sorted Array II

    Concepts:
    - array
    - two pointers

    Stats:
        Runtime | 44 ms     [Beats 95.77%]
        Memory  | 16.62 MB  [Beats 99.26%]
"""

class Solution:
    def removeDuplicates(self, nums):
        if len(nums) <= 2:
            return len(nums)

        k = 2
        for i in range(2, len(nums)):
            if nums[i] != nums[k - 2]:
                nums[k] = nums[i]
                k += 1
        return k

File: Solutions/Cpp/remove_duplicates_from_sorted_array_ii.cpp
/*
    [MEDIUM]
    80. Remove Duplicates from Sorted Array II

    Concepts:
    - array
    - two pointers

    Stats:
        Runtime | 7 ms      [Beats 54.99%]
        Memory  | 19.60 MB  [Beats 35.16%]
*/

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 2) return nums.size();
        
        int k = 2;
        for (int i = 2; i < nums.size(); i++) {
            if (nums[i] != nums[k-2]) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};